File: server/app/tables/chapter.py
import sqlalchemy

# Database table definitions.
metadata = sqlalchemy.MetaData()

chapters = sqlalchemy.Table(
    "crawler_chapters",
    metadata,
    sqlalchemy.Column("id", sqlalchemy.Integer, primary_key=True),
    sqlalchemy.Column("title", sqlalchemy.String),
    sqlalchemy.Column("content", sqlalchemy.Text),
    sqlalchemy.Column("index", sqlalchemy.Integer),
    sqlalchemy.Column("sum_words", sqlalchemy.Integer),
    sqlalchemy.Column("novel_id", sqlalchemy.Integer),
    sqlalchemy.Column("created_at", sqlalchemy.DateTime),
    sqlalchemy.Column("updated_at", sqlalchemy.DateTime),
    sqlalchemy.Column("fingerprint", sqlalchemy.String)
)

File: crawler/src/models/chapter.js
const Sequelize = require('sequelize');
const db = require('./db');

const { STRING, INTEGER, TEXT } = Sequelize.DataTypes;

module.exports = db.define(
    'crawler_chapter',
    {
        title: {
            type: STRING,
            required: true,
        },
        sum_words: {
            type: INTEGER,
            defaultValue: 0,
        },
        novel_id: {
            type: INTEGER,
            required: true,
        },
        index: {
            type: INTEGER,
            required: true,
        },
        fingerprint: {
            // title + novelId md5
            type: STRING,
            unique: true,
        },
        content: {
            type: TEXT,
        },
    },
    {
        timestamps: true,
        indexes: [
            {
                name: 'crawler_chapter_novel_id_index_index',
                method: 'BTREE',
                fields: ['novel_id', 'index'],
            },
        ],
    }
);
